src/compiler/rust/canister_methods/src/upload_file_chunk/mod.rs

```diff
@@ -0,0 +1,77 @@
+pub mod verify_latest;
+pub mod write_chunk;
+
+use quote::quote;
+
+use crate::hash_file;
+
+pub fn get_upload_file_chunk() -> proc_macro2::TokenStream {
+    let write_chunk_src = write_chunk::get_write_chunk();
+    let verify_latest_src = verify_latest::get_verify_latest();
+    let hash_file_src = hash_file::get_hash_file();
+
+    quote! {
+        #write_chunk_src
+        #verify_latest_src
+        #hash_file_src
+
+        #[ic_cdk_macros::update]
+        pub fn upload_file_chunk(
+            dest_path: String,
+            timestamp: u64,
+            start_index: u64,
+            file_bytes: Vec<u8>,
+            total_file_len: u64,
+        ) {
+            if !ic_cdk::api::is_controller(&ic_cdk::api::caller()) {
+                panic!("Must be a controller to upload files!");
+            }
+
+            let is_current_timestamp = verify_latest_version(&dest_path, timestamp);
+
+            if !is_current_timestamp {
+                return;
+            }
+
+            let uploaded_file_len = match write_chunk(&dest_path, file_bytes, start_index, total_file_len) {
+                Ok(len) => len as u64,
+                Err(err) => {
+                    ic_cdk::println!("Error writing chunk: {}", err);
+                    panic!("{}", err)
+                }
+            };
+
+            ic_cdk::println!(
```

Comment:
Do to the not quite synchronous nature of the canister side the line breaks don't really help... They might hinder more than anything because they group things up into things that are not actually groups. I'll leave them in and you can render an opinion during the final review

Reply:
We decided that on the canister end we will get rid of the line breaks

Reply:
done